internal/distro/fedora31/distro.go

```diff
@@ -396,7 +351,61 @@ func New() *Fedora31 {
 		bootable:      true,
 		defaultSize:   2 * GigaByte,
 		assembler: func(uefi bool, size uint64) *osbuild.Assembler {
-			return r.qemuAssembler("vmdk", "disk.vmdk", uefi, size)
+			return qemuAssembler("vmdk", "disk.vmdk", uefi, size)
+		},
+	}
+
+	r := Fedora31{
+		buildPackages: []string{
+			"dnf",
+			"dosfstools",
+			"e2fsprogs",
+			"policycoreutils",
+			"qemu-img",
+			"systemd",
+			"tar",
+			"xz",
+		},
+		arches: map[string]fedora31Arch{
+			"x86_64": {
+				name: "x86_64",
+				bootloaderPackages: []string{
+					"grub2-pc",
+				},
+				buildPackages: []string{
+					"grub2-pc",
+				},
+				imageTypes: map[string]imageType{
+					"ami": amiImgType,
```

Comment:
I'd use `amiImgType.name` as the key, rather than typing out the string, that way we only define the string in one place, and don't risk typos. Alternatively, add a helper that takes image types as input and returns the map (by doing the `t.Name()` lookup internally to get the keys).